### PR TITLE
User can Load Routes for a Full Trip

### PR DESCRIPTION
**Pull request by desolagiwa**

### CODE REVIEW

**Comment by desolagiwa** [comment]
This commit parses and restructures route data gotten from database for easy representation of full trips.

File: client/capstone-app/src/pages/Routes.jsx [hunk 1]
```diff
@@ -1 +1,98 @@
...
+  }, []);
+
+
+
+
+
+  return (
+    <>
+      {loading ? (
+      <div>Loading...</div>
```
Comment (slinakm) [inline client/capstone-app/src/pages/Routes.jsx:71]: Nice loading info shown to the user!

File: client/capstone-app/utils.js [hunk 2]
```diff
@@ -90,5 +90,69 @@ function midpoint(coord1, coord2) {
   return [midpointLat, midpointLon];
 }
 
+  // Helper function to flatten nested arrays
+function parseRouteData(data) {
```
Comment (slinakm) [inline client/capstone-app/utils.js:94]: Can we clarify this comment a little more? What format is the route data initially in and why are we flattening it?

File: server/routes/routes.js [hunk 3]
```diff
@@ -236,22 +235,108 @@ async function getConnectedStops(stop){
...
       }
     }
-    return currentRoutes
+    maxTransfers --
+  }
+  return currentRoutes
+}
+
+
+// This function can definitely be cleaner, but for the purpose of unblocking myself, I leaving it for now.
```
Comment (slinakm) [inline server/routes/routes.js:257]: Nice! Yeah, like you mentioned, I also see some places where we could have early returns to reduce nesting.

**Review by slinakm — APPROVED** [review]
Overall, very nice!

File: client/capstone-app/utils.js (hunk 2 above)
Comment (desolagiwa) [reply client/capstone-app/utils.js:94]: This has been addressed in future PRs! 😁

**Pull request merged**